packages/slice-machine/src/modules/pushChangesSaga/index.ts

```diff
@@ -26,24 +26,59 @@ import {
 import { LoadingKeysEnum } from "../loading/types";
 import { ApiError } from "@src/models/ApiError";
 import { SyncError } from "@src/models/SyncError";
+import { ModelStatusInformation } from "@src/hooks/useModelStatus";
+import { ModelStatus } from "@lib/models/common/ModelStatus";
+import Tracker from "@src/tracking/client";
+
+const startTimer =
+  (startTime = Date.now()) =>
+  (endTime = Date.now()) =>
+    endTime - startTime;
 
 export const changesPushCreator = createAction("PUSH_CHANGES")<{
   unSyncedSlices: ReadonlyArray<ComponentUI>;
   unSyncedCustomTypes: ReadonlyArray<CustomTypeSM>;
+  modelStatuses: ModelStatusInformation["modelsStatuses"];
   onChangesPushed: (pushed: string | null) => void;
   handleError: (e: SyncError | null) => void;
 }>();
 
 export function* changesPushSaga({
   payload,
 }: ReturnType<typeof changesPushCreator>): Generator {
-  const { unSyncedSlices, unSyncedCustomTypes, onChangesPushed, handleError } =
-    payload;
+  const {
+    unSyncedSlices,
+    unSyncedCustomTypes,
+    modelStatuses,
+    onChangesPushed,
+    handleError,
+  } = payload;
   const totalNumberOfChanges: number =
     unSyncedSlices.length + unSyncedCustomTypes.length;
 
   let alreadySyncedChanges = 0;
   let stop: Parameters<typeof handleError>[0] = null;
+  let slicesCreated = 0;
+  let slicesModified = 0;
+  const slicesDeleted = 0;
+  let customTypesCreated = 0;
+  let customTypesModified = 0;
+  const customTypesDeleted = 0;
+  let errors = 0;
+  const endTimer = startTimer();
+
+  const sendTracking = () =>
+    Tracker.get().trackChangesPushed({
+      customTypesCreated,
+      customTypesModified,
+      customTypesDeleted,
+      slicesCreated,
+      slicesModified,
+      slicesDeleted,
+      errors,
+      total: totalNumberOfChanges,
```

Comment:
I think that's only good if the entire push is a success.
Should probably be computed inside the function itself using the other parameters

Reply:
The ticket it mentioned what to do about errors, I figured we would want to know about them.